components/style/properties/longhand/ui.mako.rs

```diff
@@ -29,3 +29,38 @@ ${helpers.single_keyword("-moz-window-dragging", "default drag no-drag", product
                          gecko_enum_prefix="StyleWindowDragging",
                          animatable=False,
                          spec="None (Nonstandard Firefox-only property)")}
+
+<%helpers:longhand name="caret-color" animatable="False" products="none"
+                   spec="https://drafts.csswg.org/css-ui/#caret-color">
+    use std::fmt;
+    use style_traits::ToCss;
+    pub use self::computed_value::T as SpecifiedValue;
+
+    pub mod computed_value {
+        use cssparser::{Color, Parser};
+        use parser::{Parse, ParserContext};
+        use values::{Auto, Either};
+        use values::HasViewportPercentage;
+        use values::computed::ComputedValueAsSpecified;
+
+        no_viewport_percentage!(Color);
+        impl ComputedValueAsSpecified for Color {}
+
+        impl Parse for Color {
```

Comment:
Um, that code is [auto-generated](https://dxr.mozilla.org/servo/rev/61393b08a9a754b6833fc724bd023ce723eae5aa/components/style/properties/helpers.mako.rs#273-278) by Mako. Also, the [computed value](https://dxr.mozilla.org/servo/rev/61393b08a9a754b6833fc724bd023ce723eae5aa/components/style/values/computed/mod.rs#106) of `specified::CSSColor` is `cssparser::Color` (which would also be derived from `Either`). What happens if you define `ColorOrAuto = Either<CSSColor, Auto>` in `values/specified/mod.rs` and use `helpers.predefined_type` to generate the code?